server/libs/infra/src/search/schemas/asset.schema.ts

```diff
@@ -0,0 +1,18 @@
+import { CollectionCreateSchema } from 'typesense/lib/Typesense/Collections';
+
+export const assetSchemaVersion = 7;
+export const assetSchema: CollectionCreateSchema = {
+  name: `assets-v${assetSchemaVersion}`,
+  fields: [
+    { name: 'userId', type: 'string', facet: false },
+    { name: 'type', type: 'string', facet: true },
+    { name: 'originalPath', type: 'string', facet: false },
+    { name: 'createdAt', type: 'string', facet: false, sort: true },
+    { name: 'modifiedAt', type: 'string', facet: false },
+    { name: 'updatedAt', type: 'string', facet: false },
+    { name: 'isFavorite', type: 'bool', facet: true },
+    // { name: 'checksum', type: 'string', facet: true },
+    { name: 'tags', type: 'string[]', facet: true },
+  ],
```

Comment:
I don't think we currently have it in the database as such, but it's probably a good idea to have _all_ the exif data (that is, basically the full output from exiftool) indexed as well.

Reply:
Agreed. Need to see how to map the nested fields or if I'll have to transform the object. Basically the limitations of fields in typesense for the nested object.

Reply:
Unless we want to support filtering on them, I expect just dumping `tag: value` into a string[] should probably do the trick.

Reply:
Right I hadn't looked at nested fields yet. Looks like i should be able to define fields with dots, so like `smartInfo.tags`

Reply:
It seems easy enough and there are probably some good use cases for filtering on camera type, etc.

Reply:
Plus gps is really interesting use case, so that should be a geo point for sure.